cpp/source/deal.II/post_processor.templates.h

```diff
@@ -186,207 +193,248 @@ void SuperCapacitorPostprocessor<dim>::reset(
+  std::vector<dealii::IndexSet> index_sets(1, locally_relevant_dofs);
+  dealii::Trilinos::MPI::BlockVector relevant_solution(index_sets);
+  relevant_solution = solution;
+
   for (auto cell : dof_handler.active_cell_iterators())
   {
-    fe_values.reinit(cell);
-    this->mp_values->get_values("solid_electrical_conductivity", cell,
-                                solid_electrical_conductivity_values);
-    this->mp_values->get_values("liquid_electrical_conductivity", cell,
-                                liquid_electrical_conductivity_values);
-    this->mp_values->get_values("density", cell, density_values);
-    this->mp_values->get_values("density_of_active_material", cell,
-                                density_of_active_material_values);
-    this->mp_values->get_values("specific_surface_area", cell,
-                                specific_surface_area_values);
-    fe_values[solid_potential].get_function_gradients(
-        solution, solid_potential_gradients);
-    fe_values[liquid_potential].get_function_gradients(
-        solution, liquid_potential_gradients);
-    fe_values[solid_potential].get_function_values(solution,
-                                                   solid_potential_values);
-    fe_values[liquid_potential].get_function_values(solution,
-                                                    liquid_potential_values);
-    for (unsigned int q_point = 0; q_point < n_q_points; ++q_point)
+    if (cell->is_locally_owned())
     {
-      this->values["joule_heating"] +=
-          (solid_electrical_conductivity_values[q_point] *
-               solid_potential_gradients[q_point] *
-               solid_potential_gradients[q_point] +
-           liquid_electrical_conductivity_values[q_point] *
-               liquid_potential_gradients[q_point] *
-               liquid_potential_gradients[q_point]) *
-          fe_values.JxW(q_point);
-      this->values["volume"] += fe_values.JxW(q_point);
-      this->values["mass"] += density_values[q_point] * fe_values.JxW(q_point);
-      if (cell->material_id() == anode_electrode_material_id)
+      fe_values.reinit(cell);
+      this->mp_values->get_values("solid_electrical_conductivity", cell,
+                                  solid_electrical_conductivity_values);
+      this->mp_values->get_values("liquid_electrical_conductivity", cell,
+                                  liquid_electrical_conductivity_values);
+      this->mp_values->get_values("density", cell, density_values);
+      this->mp_values->get_values("density_of_active_material", cell,
+                                  density_of_active_material_values);
+      this->mp_values->get_values("specific_surface_area", cell,
+                                  specific_surface_area_values);
+      if (*std::max_element(solid_electrical_conductivity_values.begin(),
+                            solid_electrical_conductivity_values.end()) >
+          1e-300)
```

Comment:
Because the conductivity is 1 over the resistivity and we don't use infinity in the database. So we never get zero but 1e-307 instead.